src/main/scala/firrtl/passes/ExpandWhens.scala

```diff
@@ -90,7 +99,13 @@ object ExpandWhens extends Pass {
         case c: IsInvalid =>
           netlist(c.expr) = WInvalid
           EmptyStmt
-        case c: Attach => c
+        case Attach(_, exprs) =>
```

Comment:
So the current semantics in Firrtl are that all whens are ignored when attaching, Example 1 will indeed turn in to `attach(a, b, c)`. (Chisel errors if you try to attach conditionally, but that's technically not relevant here)